tests/functional/README-functional.md

```diff
@@ -0,0 +1,139 @@
+    proc = subprocess.run(cmd, capture_output=True, text=True, check=False)
+
+    # Step 3: Check if the test passed.
+    assert proc.returncode == 0
+```
+
+Above is an example of how a test should be written.
+To ensure that `pytest` recognizes your functions as tests, it's essential to begin the function name with `test_`.
+
+To maintain clarity in your functional tests, please provide concise and well-structured documentation explaining the **test type** (positive or negative), **rationale**, and **inputs** (if any).
+Here is a list below explaining the expected components of the documentation:
+
+- **positive test**: indicates a successful execution with the provided input parameters.
+- **negative test**: indicates that an error is supposed to be expected in the test.
+- **rational**: reason behind this test, what is it trying to emulate?
+- **inputs**: provide input documentation (e.g. explaining the dataset used).
+
+Doing so will enable other developers in the `CytoSnake` community to fully comprehend the implemented functional tests.
+
+### Understanding the main steps of CytoSnake's Functional Tests
+
+The process of creating a functional test is straightforward, involving three key steps that developers must follow to build a successful and robust test.
+These steps consist of:
+
+- **preparation**: Preparing testing folder, dataset, and selecting inputs
+- **execution**: Execute `CytoSnake` within testing older
+- **assertion checks**: Checking desired outputs
+
+#### Preparation
+
+Preparing your dataset requires minimal effort with the help  `prepare_dataset` function.
+This function readily prepares the dataset using just two parameters: the name of the dataset and the target directory for the test execution.
+
+> **Note**: Thanks to the use of pytest's fixtures, the `testing_dir` variable already holds the path to the generated testing directory. As a result, users **are not required** to create their own testing directories when conducting functional tests.
+
+The dataset names are derived from the folder names within the `cytosnake/tests/datasets` directory.
+If the desired dataset is not available, developers can create their datasets; however, they must ensure they follow `CytoSnake's` input requirements.
+
+After executing the `prepare_dataset` function, it generates an object called `DataFiles`, granting users full control over the selection of input files for `CytoSnake`.
+For instance, if the chosen dataset comprises multiple plates, users can select a specific plate using Python indexing, as demonstrated in the example above.
+Once the desired files are selected, users can proceed to prepare them for execution.
+
+#### Execution
+
+In the execution step, `CytoSnake` runs with the provided inputs within the testing directory.
+Fortunately, the `prepare_dataset` function not only generates the `DataFile` object but also informs the testing module about the execution location for `CytoSnake`.
```

Comment:
```suggestion
Fortunately, the `prepare_dataset()` function not only generates the `DataFile` object but also informs the testing module about the execution location for `CytoSnake`.
```

I'm used to seeing functions referred to like this, but please do not accept my suggestion if you are being consistent without `()`